g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductComponentAction.java

```diff
@@ -225,19 +223,31 @@ private int compare(final String source, final String target) {
 		if(!StringUtils.isEmpty(source) && StringUtils.isEmpty(target)) {
 			return 1;
 		}
-		String[] s = source.split("\\.");
+
+		String f = filterVersion(source, target);
+
+		String[] s = f.split("\\.");
 		String[] t = target.split("\\.");
-		int b = -1;
+		int b = 0;
 		if(s.length == t.length) {
 			for(int i = 0; i < s.length; i++) {
 				if(Integer.parseInt(s[i]) > Integer.parseInt(t[i])) {
 					b = 1;
 					break;
 				} else if(Integer.parseInt(s[i]) < Integer.parseInt(t[i])) {
+					b = -1;
 					break;
 				}
 			}
 		}
+
+		if(b == 0) {
+			if(source.length() < target.length()) {
```

Comment:
"9.00.0", "9.0.00" are treated equal. Is this correct?

Reply:
I think they should be treated as equal, since if we we compare them part by part, 0 should be same as 00.
"2.0.1" is greater than "2.00000" by the compare function.